client/rest-high-level/src/main/java/org/elasticsearch/client/searchable_snapshots/MountSnapshotRequest.java

```diff
@@ -115,19 +115,19 @@ public MountSnapshotRequest indexSettings(final Settings indexSettings) {
         return this;
     }
 
-    private String[] ignoredIndexSettings;
+    private String[] ignoreIndexSettings;
 
-    public String[] getIgnoredIndexSettings() {
-        return ignoredIndexSettings;
+    public String[] getIgnoreIndexSettings() {
```

Comment:
I am not sure of our breaking change guarantees for the java code of HLRC. In particular here, where it goes into a patch release. I prefer doing the rename that you have here, but I wonder if we should leave the old method in place, with `@Deprecated` and a proper javadoc warning on it to be bwc. Perhaps @elastic/es-clients has input to this.

Reply:
Just to confirm behavior, users previously were able to set `ignored_index_settings` but the parameter would have no effect. With this change the parameter will start having an effect?

If so, I think this rename and removal of the previous method is correct. Upgrading users should get an error if they were erroneously using a parameter that did nothing. This would prompt them to choose if they want the parameter to have an effect or if they actually do want the "nothing" case.

(Also for future pings we use the `@elastic/clients-team` handle now)

Reply:
> With this change the parameter will start having an effect?

Yes

I don't think we have strong breaking change guarantees on Java APIs, do we? I see this change as a bugfix and renaming the method is a way to notify that the previous behavior was buggy and requires to take a decision.

Keeping the previous method as deprecated avoids any compilation/runtime error but it does not look like fair to me to keep the bug around... If the previous method was used we can expect the user to interested into a fix for it? Also, the hassle on the breaking change in Java is easy to fix so breaking the API looks like the right thing to do (specially on a deprecated client and on a not so widely used API).

If that's OK I'll continue with the removal of the old method.

Reply:
FWIW, the breaking change policy in the context of clients is somewhat less strict than that in ES. Here's the one we wrote for [the new Java API client](https://www.elastic.co/guide/en/elasticsearch/client/java-api-client/current/breaking-changes-policy.html).

Long story short, causing a compilation error where the buggy method is used is fine. As for it happening in a patch releases, we have to consider that 7.17 is the LTS version of 7.x and patch releases happen at the same time as minor releases in 8.x.

Reply:
Thanks for clarifying and for the link to the breaking changes policy. I am good with this change too then.